traversal/scanner/GraphIterator.java

```diff
@@ -301,200 +284,146 @@ private boolean verifyLoops() {
 
         private boolean isClosure(ProcedureEdge<?, ?> edge, Vertex<?, ?> fromVertex, Vertex<?, ?> toVertex) {
             if (edge.isRolePlayer()) {
-                Scopes.Scoped scoped = scopes.getOrInitialise(edge.asRolePlayer().scope());
-                return edge.asRolePlayer().isClosure(graphMgr, fromVertex, toVertex, params, scoped);
+                Scope scope = scopes.get(edge.asRolePlayer().scope());
+                return edge.asRolePlayer().isClosure(graphMgr, fromVertex, toVertex, params, scope);
             } else {
                 return edge.isClosure(graphMgr, fromVertex, toVertex, params);
             }
         }
 
-        private void addEdgeInput(ProcedureEdge<?, ?> edge, Vertex<?, ?> fromVertex) {
-            assert !edgeInputs.containsKey(edge);
-            if (iterator != null && iterator.hasNext()) {
-                edgeInputIntersectionCache.put(new HashMap<>(edgeInputs), iterator.peek());
-                iterator = intersect(branch(fromVertex, edge), iterator);
-            }
-            edgeInputs.put(edge, fromVertex);
-            vertex = null;
-        }
-
-        private void removeEdgeInput(ProcedureEdge<?, ?> edge) {
-            edgeInputIntersectionCache.remove(edgeInputs);
-            edgeInputs.remove(edge);
-            reset();
-        }
-
-        public void clearCurrentVertex() {
-            vertex = null;
-        }
-
-        private void clear() {
-            edgeInputs.clear();
-            edgeInputIntersectionCache.clear();
-            reset();
-        }
-
-        private void reset() {
+        void clear() {
             if (iterator != null) {
                 iterator.recycle();
                 iterator = null;
             }
+            clearCurrentVertex();
+            if (procedureVertex.id().isScoped()) localScope.remove(procedureVertex);
+            else {
+                iterate(procedureVertex.ins()).filter(ProcedureEdge::isRolePlayer).forEachRemaining(e -> {
+                    Scope scope = scopes.get(e.asRolePlayer().scope());
+                    scope.remove(e);
+                });
+            }
+        }
```

Comment:
`reset()`clears the iterator and the vertex, along with unscoping any scoped roles.

Reply:
I assume you meant `clear()`